core/src/test/scala/org/apache/spark/sql/delta/cdc/MergeCDCSuite.scala

```diff
@@ -187,6 +193,118 @@ class MergeCDCSuite extends MergeIntoSQLSuite with DeltaColumnMappingTestUtils {
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with duplicate matches with duplicates in the target")(
+    target = Seq(0, 1, 1).toDF("value"),
+    source = Seq(1, 1).toDF("value"),
+    mergeCondition = "t.value = s.value",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: (1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with target-only condition")(
+    target = Seq(0, 1).toDF("value"),
+    source = Seq(0, 1).toDF("value"),
+    mergeCondition = "t.value > 0",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with target-only condition with duplicates in the target")(
+    target = Seq(0, 1, 1).toDF("value"),
+    source = Seq(0, 1).toDF("value"),
+    mergeCondition = "t.value > 0",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData = Seq(0).toDF(),
+    expectedCdcData = ((1, "delete", 1) :: (1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with source-only condition")(
+    target = Seq(0, 1).toDF("value"),
+    source = Seq(0, 1).toDF("value"),
+    mergeCondition = "s.value < 2",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData =
+      Nil.asInstanceOf[List[Integer]]
+      .toDF("value"),
+    expectedCdcData = ((0, "delete", 1) :: (1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with source-only condition with duplicates in the target")(
+    target = Seq(0, 1, 1).toDF("value"),
+    source = Seq(0, 1).toDF("value"),
+    mergeCondition = "s.value < 2",
+    clauses = MergeClause(isMatched = true, null, "DELETE") :: Nil,
+    expectedTableData =
+      Nil.asInstanceOf[List[Integer]]
+        .toDF("value"),
+    expectedCdcData = ((0, "delete", 1) :: (1, "delete", 1) :: (1, "delete", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses("unconditional delete only with duplicate matches + insert")(
+    target = ((1, 1) :: (2, 2) :: Nil).toDF("key", "value"),
+    source = ((1, 10) :: (1, 100) :: (3, 30) :: (3, 300) :: Nil).toDF("key", "value"),
+    mergeCondition = "s.key = t.key",
+    clauses = MergeClause(isMatched = true, null, "DELETE") ::
+      insert(values = "(key, value) VALUES (s.key, s.value)") :: Nil,
+    expectedTableData = ((2, 2) :: (3, 30) :: (3, 300) :: Nil).toDF("key", "value"),
+    expectedCdcData =
+      ((1, 1, "delete", 1) :: (3, 30, "insert", 1) :: (3, 300, "insert", 1) :: Nil).toDF()
+  )
+
+  testMergeCdcUnlimitedClauses(
+    "unconditional delete only with duplicate matches + insert with duplicate matches")(
```

Comment:
insert with duplicate match does not make sense. insert with duplicate keys.

Reply:
duplicate rows? does that work? It's not just the keys that are duplicated.. its the `(3, 300) :: (3, 300)`